src/StackExchange.Redis/VectorSetSimilaritySearchMessage.cs

```diff
@@ -0,0 +1,322 @@
+        }
+
+        if (HasFlag(VsimFlags.WithScores)) argCount++; // [WITHSCORES]
+        if (HasFlag(VsimFlags.WithAttributes)) argCount++; // [WITHATTRIBS]
+        if (HasFlag(VsimFlags.Count)) argCount += 2; // [COUNT {count}]
+        if (HasFlag(VsimFlags.Epsilon)) argCount += 2; // [EPSILON {epsilon}]
+        if (HasFlag(VsimFlags.SearchExplorationFactor)) argCount += 2; // [EF {search-exploration-factor}]
+        if (filterExpression is not null) argCount += 2; // [FILTER {filterExpression}]
+        if (HasFlag(VsimFlags.MaxFilteringEffort)) argCount += 2; // [FILTER-EF {max-filtering-effort}]
+        if (HasFlag(VsimFlags.UseExactSearch)) argCount++; // [TRUTH]
+        if (HasFlag(VsimFlags.DisableThreading)) argCount++; // [NOTHREAD]
+        return argCount;
+    }
+
+    protected override void WriteImpl(PhysicalConnection physical)
+    {
+        var useFp32 = VectorSetAddMessage.UseFp32; // avoid race in debug mode
+        physical.WriteHeader(Command, GetArgCount(useFp32));
+
+        // Write key
+        physical.Write(key);
+
+        // Write search target: either "ELE {member}" or vector data
+        if (!member.IsNull)
+        {
+            // Member-based search: "ELE {member}"
+            physical.WriteBulkString("ELE"u8);
+            physical.WriteBulkString(member);
+        }
+        else
+        {
+            // Vector-based search: either "FP32 {vector}" or "VALUES {num} {vector}"
+            if (useFp32)
+            {
+                physical.WriteBulkString("FP32"u8);
+                physical.WriteBulkString(System.Runtime.InteropServices.MemoryMarshal.AsBytes(vector.Span));
+            }
+            else
+            {
+                physical.WriteBulkString("VALUES"u8);
+                physical.WriteBulkString(vector.Length);
+                foreach (var val in vector.Span)
+                {
+                    physical.WriteBulkString(val);
+                }
+            }
+        }
+
+        if (HasFlag(VsimFlags.WithScores))
+        {
+            physical.WriteBulkString("WITHSCORES"u8);
+        }
+
+        if (HasFlag(VsimFlags.WithAttributes))
+        {
+            physical.WriteBulkString("WITHATTRIBS"u8);
+        }
+
+        // Write optional parameters
+        if (HasFlag(VsimFlags.Count))
+        {
+            physical.WriteBulkString("COUNT"u8);
+            physical.WriteBulkString(_count);
+        }
+
+        if (HasFlag(VsimFlags.Epsilon))
+        {
+            physical.WriteBulkString("EPSILON"u8);
+            physical.WriteBulkString(_epsilon);
+        }
+
+        if (HasFlag(VsimFlags.SearchExplorationFactor))
+        {
+            physical.WriteBulkString("EF"u8);
+            physical.WriteBulkString(_searchExplorationFactor);
+        }
+
+        if (filterExpression is not null)
+        {
+            physical.WriteBulkString("FILTER"u8);
+            physical.WriteBulkString(filterExpression);
```

Comment:
Let's check in tests for empty string here to be sure that behaves correctly

Reply:
applying a null-or-whitespace filter instead, with integration tests for null, empty, whitespace, and non-whitespace values